pkg/querier/tripperware/instantquery/instant_query.go

```diff
@@ -324,6 +335,59 @@ func vectorMerge(resps []*PrometheusInstantQueryResponse) *Vector {
 	return result
 }
 
+func matrixMerge(resps []*PrometheusInstantQueryResponse) *Matrix {
```

Comment:
This function looks a lot like https://github.com/cortexproject/cortex/blob/997e4af542e8363b02ddb759bad4019cef9d1692/pkg/querier/tripperware/queryrange/query_range.go#L347 

Anyway we can reduce the duplication?

Reply:
No good way for now cause they are different types.

Reply:
We could expose an interface for sure... But this needs more refactoring. I prefer copying the function for now.

Reply:
From my brief look, I think we can break https://github.com/cortexproject/cortex/blob/997e4af542e8363b02ddb759bad4019cef9d1692/pkg/querier/tripperware/queryrange/query_range.go#L347 into multiple functions one that takes in `[]tripperware.SampleStream` and the other that takes in `resps []*PrometheusResponse`. 

Then here you may be able to use the new function that takes in `[]tripperware.SampleStream`.

Reply:
Updated. Could you please take another look?